File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/helpers/AprilTagConfig.java
package org.firstinspires.ftc.teamcode.helpers;

import org.firstinspires.ftc.teamcode.Helper.Hardware;
import org.firstinspires.ftc.vision.VisionPortal;
import org.firstinspires.ftc.vision.apriltag.AprilTagDetection;
import org.firstinspires.ftc.vision.apriltag.AprilTagProcessor;

import java.util.Optional;

public class AprilTagConfig {
    private final AprilTagProcessor aprilTag;
    private final VisionPortal visionPortal;
    public Optional<BallOrder> order = Optional.empty();
    public AprilTagConfig(){
        aprilTag = AprilTagProcessor.easyCreateWithDefaults();
        visionPortal = VisionPortal.easyCreateWithDefaults(Hardware.camera, aprilTag);
    }
    public void Update(){
        if(!order.isPresent()) {
            for (AprilTagDetection d : aprilTag.getDetections()) {
                switch (d.id) {
                    case 21:
                        order = Optional.of(BallOrder.GPP);
                        break;
                    case 22:
                        order =  Optional.of(BallOrder.PGP);
                        break;
                    case 23:
                        order =  Optional.of(BallOrder.PPG);
                        break;
                }
            }
        }
    }
    public void Close(){
        visionPortal.close();
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/helper/AprilTagConfig.kt
package org.firstinspires.ftc.teamcode.helper

import org.firstinspires.ftc.vision.VisionPortal
import org.firstinspires.ftc.vision.apriltag.AprilTagProcessor

class AprilTagConfig {
    private val aprilTag: AprilTagProcessor = AprilTagProcessor.easyCreateWithDefaults()

    private val visionPortal: VisionPortal =
        VisionPortal.easyCreateWithDefaults(Hardware.camera, aprilTag)
    var order: BallOrder? = null

    fun update() {
        if (order == null) {
            for (d in aprilTag.detections) {
                order = when (d.id) {
                    21 -> BallOrder.GPP
                    22 -> BallOrder.PGP
                    23 -> BallOrder.PPG
                    else -> null
                }
            }
        }
    }

    fun close() {
        visionPortal.close()
    }
}
